lxd/images.go

```diff
@@ -597,57 +433,52 @@ func dbInsertImage(d *Daemon, fp string, fname string, sz int64, public int,
 	return nil
 }
 
-func imagesPost(d *Daemon, r *http.Request) Response {
-	dirname := shared.VarPath("images")
-	if err := os.MkdirAll(dirname, 0700); err != nil {
-		return InternalError(err)
-	}
-
-	// create a directory under which we keep everything while building
-	builddir, err := ioutil.TempDir(dirname, "lxd_build_")
+func imageBuildFromInfo(d *Daemon, info shared.ImageInfo) (metadata map[string]string, err error) {
+	err = d.Storage.ImageCreate(info.Fingerprint)
 	if err != nil {
-		return InternalError(err)
+		return metadata, err
 	}
 
-	/* remove the builddir when done */
-	defer removeImgWorkdir(d, builddir)
-
-	/* Grab all info from the web request */
-
-	info, err := getImgPostInfo(d, r, builddir)
+	err = dbInsertImage(
+		d,
+		info.Fingerprint,
+		info.Filename,
+		info.Size,
+		info.Public,
+		info.Architecture,
+		info.CreationDate,
+		info.ExpiryDate,
+		info.Properties)
 	if err != nil {
-		return SmartError(err)
+		return metadata, err
 	}
 
-	metadata, err := buildImageFromInfo(d, info, builddir)
-	if err != nil {
-		return SmartError(err)
-	}
-	return SyncResponse(true, metadata)
+	metadata = make(map[string]string)
+	metadata["fingerprint"] = info.Fingerprint
+	metadata["size"] = strconv.FormatInt(info.Size, 10)
+
+	return metadata, nil
 }
 
-func buildImageFromInfo(d *Daemon, info shared.ImageInfo, builddir string) (metadata map[string]string, err error) {
-	if err := buildOtherFs(d, builddir, info.Fingerprint); err != nil {
-		return nil, err
+func imagesPost(d *Daemon, r *http.Request) Response {
+	dirname := shared.VarPath("images")
+	if err := os.MkdirAll(dirname, 0700); err != nil {
+		return InternalError(err)
```

Comment:
I know you're just moving existing code, but those MkdirAll for the default paths:
 - /var/lib/lxd/devlxd
 - /var/lib/lxd/images
 - /var/lib/lxd/lxc

Seems like they should just all be created if missing when the daemon first starts, not in every function that may use the path.